frontend/lib/server-utils.ts

```
@@ -0,0 +1,21 @@
+import fs from "fs";
+import path from "path";
+
+export const getEmailsConfig = async (): Promise<string[] | false> => {
+  try {
+    const filePath = path.join(process.cwd(), "allowed-emails.json");
+
+    if (!fs.existsSync(filePath)) {
+      console.log("file not found :(");
+      return false;
+    }
+
+    const fileContent = fs.readFileSync(filePath, "utf-8");
+
+    const jsonData = JSON.parse(fileContent) as { emails: string[] };
+    return jsonData.emails;
```

Even though this may throw TS errors, I think it will silently return `undefined` for valid JSON files with invalid format. Please handle, by at least returning `jsonData.emails ?? []` or better throw an error if `jsonData.emails === undefined`